internal/cmd/config.go

```diff
@@ -1074,64 +1075,64 @@ func (c *Config) defaultPreApplyFunc(
 		return nil
 	}
 
-	if c.Interactive {
-		prompt := fmt.Sprintf("Apply %s", targetRelPath)
-		var choices []string
-		actualContents := actualEntryState.Contents()
-		targetContents := targetEntryState.Contents()
-		if actualContents != nil || targetContents != nil {
-			choices = append(choices, "diff")
+	// Prepare decision for which kind of prompt we need (if any)
+	type promptMode int
+	const (
+		promptNone     promptMode = iota
+		promptYesNoAll            // yes/no/all/quit (just ask, don't indicate if there is a conflict)
+		promptConflict            // overwrite/all-overwrite/skip/quit (conflict-specific prompt)
+	)
+	mode := promptNone
+
+	targetDirty := lastWrittenEntryState != nil && !lastWrittenEntryState.Equivalent(actualEntryState)
+	targetPreExisting := lastWrittenEntryState == nil && actualEntryState.Type != chezmoi.EntryStateTypeRemove
+
+	// Select prompt mode based on command line flag
+	switch {
+	case c.Interactive: // Prompt no matter what
+		mode = promptYesNoAll
+	case c.LessInteractive: // Prompt if target is dirty or pre-existing (i.e., only overwrite what chezmoi has written)
+		if targetDirty || targetPreExisting {
+			mode = promptConflict
 		}
-		choices = append(choices, choicesYesNoAllQuit...)
-		for {
-			switch choice, err := c.promptChoice(prompt, choices); {
-			case err != nil:
-				return err
-			case choice == "diff":
-				err := c.diffFile(
-					targetRelPath,
-					c.DestDirAbsPath.Join(targetRelPath), actualContents, actualEntryState.Mode,
-					chezmoi.EmptyAbsPath, targetContents, targetEntryState.Mode,
-				)
-				if err != nil {
-					return err
-				}
-			case choice == "yes":
-				return nil
-			case choice == "no":
-				return fs.SkipDir
-			case choice == "all":
-				c.Interactive = false
-				return nil
-			case choice == "quit":
-				return chezmoi.ExitCodeError(1)
-			default:
-				panic(choice + ": unexpected choice")
-			}
+	default: // Prompt in *some* cases of a dirty target:
+		switch {
+		case targetEntryState.Overwrite():
+		case targetEntryState.Type == chezmoi.EntryStateTypeScript:
+		case lastWrittenEntryState == nil:
```

Comment:
In Go, `switch` cases `break` by default unless you add an explicit `fallthrough` statement. This unlike C or C++ where `switch` cases fall through by default unless you add an explicit `break` statement. So, most of the cases in this `switch` have no effect. Is this correct?